src/components/ble/SimpleWeatherService.h

```diff
@@ -0,0 +1,140 @@
+/*  Copyright (C) 2023 Jean-François Milants
+
+    This file is part of InfiniTime.
+
+    InfiniTime is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    InfiniTime is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <memory>
+
+#define min // workaround: nimble's min/max macros conflict with libstdc++
+#define max
+#include <host/ble_gap.h>
+#include <host/ble_uuid.h>
+#include <optional>
+#include <cstring>
+#undef max
+#undef min
+
+#include "components/datetime/DateTimeController.h"
+
+int WeatherCallback(uint16_t connHandle, uint16_t attrHandle, struct ble_gatt_access_ctxt* ctxt, void* arg);
+
+namespace Pinetime {
+  namespace Controllers {
+
+    class SimpleWeatherService {
+    public:
+      explicit SimpleWeatherService(const DateTime& dateTimeController);
+
+      void Init();
+
+      int OnCommand(struct ble_gatt_access_ctxt* ctxt);
+
+      static constexpr uint8_t MaxNbForecastDays = 5;
+
+      enum class Icons : uint8_t {
+        Sun = 0,       // ClearSky
+        CloudsSun = 1, // FewClouds
+        Clouds = 2,    // Scattered clouds
+        BrokenClouds = 3,
+        CloudShowerHeavy = 4, // shower rain
+        CloudSunRain = 5,     // rain
+        Thunderstorm = 6,
+        Snow = 7,
+        Smog = 8, // Mist
+        Unknown = 255
+      };
+
+      struct CurrentWeather {
+        CurrentWeather(uint64_t timestamp,
+                       uint8_t temperature,
+                       uint8_t minTemperature,
+                       uint8_t maxTemperature,
+                       uint8_t iconId,
+                       const char* location)
+          : timestamp {timestamp},
+            temperature {temperature},
+            minTemperature {minTemperature},
+            maxTemperature {maxTemperature},
+            iconId {iconId} {
+          std::memcpy(this->location, location, 32);
+          this->location[32] = 0;
+        }
+
+        uint64_t timestamp;
+        uint8_t temperature;
+        uint8_t minTemperature;
+        uint8_t maxTemperature;
+        Icons iconId;
+        char location[33]; // 32 char + \0 (end of string)
+
+        bool operator==(const CurrentWeather& other) const;
+      };
+
+      struct Forecast {
+        uint64_t timestamp;
+        uint8_t nbDays;
+
+        struct Day {
+          uint8_t minTemperature;
+          uint8_t maxTemperature;
+          uint8_t iconId;
```

Comment:
This should probably be an `Icons`.

```suggestion
          Icons iconId;
```

Reply:
Done